src/main/java/org/opensearch/knn/plugin/transport/ClearCacheTransportAction.java

```diff
@@ -0,0 +1,168 @@
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.core.action.support.DefaultShardOperationFailedException;
+import org.opensearch.action.support.broadcast.node.TransportBroadcastByNodeAction;
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.block.ClusterBlockException;
+import org.opensearch.cluster.block.ClusterBlockLevel;
+import org.opensearch.cluster.metadata.IndexNameExpressionResolver;
+import org.opensearch.cluster.routing.ShardRouting;
+import org.opensearch.cluster.routing.ShardsIterator;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.index.Index;
+import org.opensearch.index.IndexService;
+import org.opensearch.index.shard.IndexShard;
+import org.opensearch.indices.IndicesService;
+import org.opensearch.knn.index.KNNIndexShard;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.TransportService;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Transport Action to evict k-NN indices from Cache. TransportBroadcastByNodeAction will distribute the request to
+ * all shards across the cluster for the given indices. For each shard, shardOperation will be called and the
+ * indices will be cleared from cache.
+ */
+public class ClearCacheTransportAction extends TransportBroadcastByNodeAction<
+    ClearCacheRequest,
+    ClearCacheResponse,
+    TransportBroadcastByNodeAction.EmptyResult> {
+
+    public static Logger logger = LogManager.getLogger(ClearCacheTransportAction.class);
+
+    private IndicesService indicesService;
+
+    /**
+     * Constructor
+     *
+     * @param clusterService ClusterService
+     * @param transportService TransportService
+     * @param actionFilters ActionFilters
+     * @param indexNameExpressionResolver IndexNameExpressionResolver
+     * @param indicesService IndicesService
+     */
+    @Inject
+    public ClearCacheTransportAction(
+        ClusterService clusterService,
+        TransportService transportService,
+        ActionFilters actionFilters,
+        IndexNameExpressionResolver indexNameExpressionResolver,
+        IndicesService indicesService
+    ) {
+        super(
+            ClearCacheAction.NAME,
+            clusterService,
+            transportService,
+            actionFilters,
+            indexNameExpressionResolver,
+            ClearCacheRequest::new,
+            ThreadPool.Names.SEARCH
+        );
+        this.indicesService = indicesService;
+    }
+
+    /**
+     * @param streamInput StreamInput
+     * @return EmptyResult
+     * @throws IOException
+     */
+    @Override
+    protected EmptyResult readShardResult(StreamInput streamInput) throws IOException {
+        return EmptyResult.readEmptyResultFrom(streamInput);
```

Comment:
Because we will not explicitly return any response from shard. If there are any exceptions, they will be thrown which will be aggregated by the BroadcastTransportAction and returned in the response.